crates/biome_js_analyze/src/analyzers/nursery/no_useless_lone_block_statements.rs

```diff
@@ -0,0 +1,125 @@
+use crate::semantic_services::Semantic;
+use biome_analyze::{
+    context::RuleContext, declare_rule, Rule, RuleDiagnostic,
+};
+use biome_console::markup;
+use biome_js_syntax::{JsBlockStatement, JsLabeledStatement, JsStatementList, AnyJsStatement, JsFunctionDeclaration, JsVariableStatement};
+use biome_js_syntax::JsVariableKind::{Const, Let};
+use biome_rowan::{AstNode, AstNodeList, TextRange};
+
+declare_rule! {
+    /// Disallow unnecessary nested block statements.
+    ///
+    /// > In JavaScript, prior to ES6, standalone code blocks delimited by curly braces do not create a new scope and have no use.
+    /// > In ES6, code blocks may create a new scope if a block-level binding (let and const), a class declaration or a function declaration (in strict mode) are present. A block is not considered redundant in these cases.
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/no-lone-blocks
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// {}
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (foo) {
+    ///   bar();
+    ///   {
+    ///     baz();
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// while (foo) {
+    ///   bar();
+    /// }
+    /// ```
+    ///
+    pub(crate) NoUselessLoneBlockStatements {
+        version: "next",
+        name: "noUselessLoneBlockStatements",
+        recommended: false,
+    }
+}
+
+impl Rule for NoUselessLoneBlockStatements {
+    type Query = Semantic<JsBlockStatement>;
+    type State = TextRange;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let block = ctx.query();
+
+        if let Some(_labeled_statement) = JsLabeledStatement::cast(block.syntax().parent()?) {
+            return None;
+        }
+
+        if block.statements().is_empty() {
+            return Some(block.range());
+        }
+
+        if block_has_control_structure(&block) {
+            return None;
+        }
+
+        for statement in block.statements() {
+            return match statement {
```

Comment:
The `return` seems an error?

Instead of a loop you could use `find_map`:

```rs
block.statements().iter().find_map(|statement| match { ... })
```

Reply:
I've ended up using `iter().any()` instead, let me know if this is ok